nvi

```diff
@@ -10,7 +10,12 @@ N=$(tput sgr0) # Normal
 U=$(tput smul) # Underline
 
 print_usage() {
-  cat << EOF
+  less \
+    --raw-control-chars \
+    --no-init \
+    --quit-if-one-screen \
+    --QUIT-AT-EOF \
```

Comment:
That's a good observation. The [GNU Coding Standards](https://www.gnu.org/prep/standards/standards.html#g_t_002d_002dhelp) is the closest thing to formal documentation of the behaviour I can find. It says very little but it does say to print to stdout like your first set of examples (but then, GNU also thinks that `info` is useful).

Personally, I never use `--help` precisely because the experience sucks donkey balls, even with a terminal multiplexer and convenient scrolling. I suppose my perspective is that requiring me to manually pipe to a pager is Unix SRP wankery. I use `man` directly, and fume when package authors only supply `--help` and no `man` page.

`git help`'s default behaviour is to invoke `man`, so `git help commit` and `man git commit` are identical. That's important because providing `man` pages is a little painful, even generated from something like AsciiDoc like Git does, and the complexity does not seem worthwhile for this tool's intended use case (many images don't even have `man`, and some distros don't include it out of the box, although a host system without it is hard to use).

On the other hand, `nvi --help` is a faux `man` page: the documentation layout and density is, by design, a fully fledged `man` page, and far more advanced than the summary produced by most `--help`s. `nvi --help` is good documentation that, puristically, doesn't belong in `--help`.

I guess you have to decide which convention you'd rather violate. We could also write an actual `man` page (it's entirely doable and I wouldn't mind doing it -- `man` can even read local files so it's not necessary to install them) and change `--help` to be a summary. In any case, we can avoid this particular misuse of `cat`.